ADR/0033-product-version-controller.md

```diff
@@ -0,0 +1,252 @@
+user put it:
+
+> As an architect, I need to be able to see all my apps versions in
+> one place, so that they are easily accessible by everyone on the
+> team and organized in a way that matches the way our team talks
+> about them, making my workflow for each app more streamlined.
+
+Some Konflux users have already attempted managing large multi-versioned
+products in Konflux. The pattern that emerged for doing so is as follows:
+
+* Create a separate Konflux Application for each product version
+* Use branches to track different source code versions within Git repositories
+* Use a separate Konflux Components for each buildable piece of code
+  (E.g. Dockerfile) in each branch in each repo that is included in a particular
+  application version.
+* To manage large amounts of components, some users have used multiple
+  application objects for a single product version.
+
+As one can expect, creating and managing so many Konflux objects is currently
+rather cumbersome. There are a few redundant steps that need to be taken. For
+example, access credentials for a Git repository need to be filled-in every time
+a component referring to that repository is created, even if other components
+referring that repository already exist in the user's workspace.
+
+The intention for the solution proposed in this document is to follow the
+pattern described above as far as the Konflux "Application" and "Component"
+objects go and provide a higher-level layer for the streamlined management of
+large amount of such objects.
+
+An alternative approach to managing the challenges we had described above is to
+modify the way Application and Component object function so that for example:
+
+* A component may encompass multiple versions of similar code across different
+  Git branches
+* An application may include several sub groups of components so that each could
+  be regarded to be defining a different version
+
+We see several benefits to the "hight level layer" approach we propose over the
+alternative:
+
+* The way Components and Applications work is currently tightly coupled to the
+  way the various Konflux services work. Changing it would require changes
+  across multiple services and may introduce operational risks.
+* There are proposals being discussed for decoupling the various Konflux
+  services and possibly deprecating the Component and Application objects.
+  Having a separate hight-level layer would allow for adjusting it for using
+  different underlying objects while continued emphasis on using Application
+  and Component object would necessitate continued maintenance of them.
+
+## Decision
+
+### Facilitating product and version based navigation in the UI
+
+We will define two new custom resources:
+
+**Project** resources will define a large project that can have multiple
+parallel development streams that may be each released as a different product
+version.
+
+**ProjectDevStream** resources will define a stream of development within a
```

Comment:
I would advocate for not using an abbreviation here...but use  `ProjectDevelopmentStream`

Reply:
Done.